databricks_blueprints/download_file_from_dbfs.py

```diff
@@ -0,0 +1,132 @@
+    parser.add_argument('--source-file-name-match-type', dest='source_file_name_match_type',
+                        choices={'exact_match', 'regex_match'}, required=True)
+    args = parser.parse_args()
+    return args
+
+
+def download_file_from_dbfs(client, source_file_path, dest_file_path):
+    read_handle_endpoint = "/dbfs/read"
+    # check if filetype works with binary or not
+    if ('.csv' in source_file_path) or ('.txt' in source_file_path):
+        is_binary = False
+        write_mode = "w+"
+    else:
+        is_binary = True
+        write_mode = 'wb+'
+    with open(dest_file_path, write_mode) as file:
+        offset = 0
+        while True:
+            payload = {
+                "path": source_file_path,
+                "offset": offset, 
+                "length": ONE_MB # 1MB of data 
+            }
+            try:
+                read_response = client.get(read_handle_endpoint, params=payload)
+            except Exception:
+                print(f'Connection Error for {read_handle_endpoint}')
+                print(f"Data downloaded: {offset/1024}MB")
+                sys.exit(errors.EXIT_CODE_UNKNOWN_ERROR)
+
+            if read_response.status_code == requests.codes.ok:
+                response_data = read_response.json()
+                data = base64.b64decode(response_data['data'])
+                if is_binary:
+                    file.write(data)
+                else:
+                    file.write(data.decode('utf-8'))
+                # check if there is still more data left to read
+                if response_data['bytes_read'] == ONE_MB:
+                    offset += ONE_MB
+                else: # that was the last of the data in the file
+                    break
+            elif read_response.status_code == 404:
+                print(f"Error: No file located at {source_file_path}")
+                sys.exit(errors.EXIT_CODE_DBFS_INVALID_FILEPATH)
+            elif read_response.status_code == 400: # Bad Request
+                error = read_response.json()
+                code = error['error_code']
+                message = error['message']
+                print(f"Error: {code} {message}")
+                sys.exit(errors.EXIT_CODE_DBFS_READ_ERROR)
+            else:
+                print(f"Failed to read data: {read_response.status_code} {read_response.text}")
+                sys.exit(errors.EXIT_CODE_DBFS_READ_ERROR)
+    print(f"finished downloading file:{source_file_path} as {dest_file_path}")
+    
+
+def main():
+    args = get_args()
+    access_token = args.access_token
+    instance_id = args.instance_id
+    source_file_name = args.source_file_name
+    source_folder_name = args.source_folder_name
+    dest_file_name = args.dest_file_name
+    dest_folder_name = args.dest_folder_name
+    source_file_name_match_type = args.source_file_name_match_type
+    # create client
+    client = helpers.DatabricksClient(access_token, instance_id)
```

Comment:
Still need to error handle somewhere for an incorrect instance id. Currently just returns:

Error sending request to /dbfs/list: HTTPSConnectionPool(host='randomid.cloud.databricks.comm', port=443): Max retries exceeded with url: /api/2.0/dbfs/list?path=%2FFileStore%2F (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x1043990a0>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known'))